antiwindupcontroller.go

```diff
@@ -73,12 +77,14 @@ func (c *AntiWindupController) Reset() {
 func (c *AntiWindupController) Update(input AntiWindupControllerInput) {
 	e := input.ReferenceSignal - input.ActualSignal
 	controlErrorIntegral := c.State.ControlErrorIntegrand*input.SamplingInterval.Seconds() + c.State.ControlErrorIntegral
+	controlErrorIntegral = math.Max(-math.MaxFloat64, math.Min(math.MaxFloat64, controlErrorIntegral))
 	controlErrorDerivative := ((1/c.Config.LowPassTimeConstant.Seconds())*(e-c.State.ControlError) +
 		c.State.ControlErrorDerivative) / (input.SamplingInterval.Seconds()/c.Config.LowPassTimeConstant.Seconds() + 1)
 	u := e*c.Config.ProportionalGain + c.Config.IntegralGain*controlErrorIntegral +
 		c.Config.DerivativeGain*controlErrorDerivative + input.FeedForwardSignal
-	c.State.ControlSignal = math.Max(c.Config.MinOutput, math.Min(c.Config.MaxOutput, u))
-	c.State.ControlErrorIntegrand = e + c.Config.AntiWindUpGain*(c.State.ControlSignal-u)
+	c.State.UnsaturatedControlSignal = u
+	c.State.ControlSignal = math.Max(c.Config.MinOutput, math.Min(c.Config.MaxOutput, c.State.UnsaturatedControlSignal))
+	c.State.ControlErrorIntegrand = e + c.Config.AntiWindUpGain*(c.State.ControlSignal-c.State.UnsaturatedControlSignal)
```

Comment:
Is it possible to remove the temporary variable `u` and assign the `UnsaturatedControlSignal` directly?
```
c.State.UnsaturatedControlSignal = e*c.Config.ProportionalGain + c.Config.IntegralGain*controlErrorIntegral +
		c.Config.DerivativeGain*controlErrorDerivative + input.FeedForwardSignal
```

Reply:
Yes, fixed that and refactored the tracking controller as well.